presto-cli/src/test/java/io/prestosql/cli/TestInsecureQueryRunner.java

```diff
@@ -0,0 +1,190 @@
+import io.prestosql.client.QueryResults;
+import io.prestosql.client.StatementStats;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import java.io.FileInputStream;
+import java.io.PrintStream;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.time.ZoneId;
+import java.util.Locale;
+import java.util.Optional;
+
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.HttpHeaders.LOCATION;
+import static com.google.common.net.HttpHeaders.SET_COOKIE;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.prestosql.cli.ClientOptions.OutputFormat.CSV;
+import static io.prestosql.client.ClientStandardTypes.BIGINT;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestInsecureQueryRunner
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+
+    private MockWebServer server;
+
+    @BeforeMethod
+    public void setup() throws Exception
+    {
+        server = new MockWebServer();
+        SSLContext sslContext = buildTestSslContext();
+        server.useHttps(sslContext.getSocketFactory(), false);
+        server.start();
+    }
+
+    private SSLContext buildTestSslContext() throws Exception
+    {
+        // Load self-signed certificate
+        String path = getClass().getClassLoader().getResource("insecure-ssl-test.jks").getPath();
+        FileInputStream stream = new FileInputStream(path);
+        char[] serverKeyStorePassword = "insecure-ssl-test".toCharArray();
+        KeyStore serverKeyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        serverKeyStore.load(stream, serverKeyStorePassword);
+
+        String kmfAlgorithm = KeyManagerFactory.getDefaultAlgorithm();
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(kmfAlgorithm);
+        kmf.init(serverKeyStore, serverKeyStorePassword);
+
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(kmfAlgorithm);
+        trustManagerFactory.init(serverKeyStore);
+        SSLContext sslContext = SSLContext.getInstance("SSL");
+        sslContext.init(kmf.getKeyManagers(), trustManagerFactory.getTrustManagers(), new SecureRandom());
+        return sslContext;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown() throws Exception
+    {
+        server.close();
+    }
+
+    @Test
+    public void testInsecureConnection() throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
```

Comment:
Let's get rid of the cookie and simplify this to use a `200` response. The test this was copied from was testing cookies, but I think that distracts from the purpose of this test.